logstash-core/lib/logstash/api/modules/settings.rb

```diff
@@ -0,0 +1,38 @@
+require 'json'
+
+# encoding: utf-8
+module LogStash
+  module Api
+    module Modules
+      class Settings < ::LogStash::Api::Modules::Base
+        def handle_logging(logging_keys, settings)
+          logging_keys.each do |key|
+            level = settings.delete(key)
+            path = key.split("logger.")[1]
+            if level.nil?
```

Comment:
right. there is a slight mixture of concerns here. `configure_logging` will throw a NPE anyways. I will remove this check and make the overall catch be more descriptive